src/shared/analytics/shared/formDataToAnalytics.ts

```diff
@@ -0,0 +1,130 @@
+    quote,
+  }: {
+    currency: string;
+    formData: AnalyticsFormData;
+    quote: Quote;
+  }
+) {
+  const spendAssetInfo = await fetchAssetFullInfo({
+    fungibleId: spendAsset.asset_code,
+    currency,
+  });
+  const receiveAssetInfo = await fetchAssetFullInfo({
+    fungibleId: receiveAsset.asset_code,
+    currency,
+  });
+
+  const fdvAssetSent = spendAssetInfo.data.fungible.meta.fullyDilutedValuation;
+  const fdvAssetReceived =
+    receiveAssetInfo.data.fungible.meta.fullyDilutedValuation;
+
+  const zerion_fee_percentage = quote.protocol_fee;
+  const feeAmount = quote.protocol_fee_amount;
+  const inputChain = createChain(quote.input_chain);
+  const outputChain = createChain(quote.output_chain);
+  const zerion_fee_usd_amount = assetQuantityToValue(
+    { quantity: feeAmount, asset: spendAsset },
+    inputChain
+  );
+  const usdAmountSend = assetQuantityToValue(
+    { quantity: quote.input_amount_estimation, asset: spendAsset },
+    inputChain
+  );
+  const usdAmountReceived = assetQuantityToValue(
+    { quantity: quote.output_amount_estimation, asset: receiveAsset },
+    outputChain
+  );
+
+  const inputValue = quote.input_amount_estimation;
+  const outputValue = quote.output_amount_estimation;
+
+  const enough_balance = new BigNumber(spendPosition?.quantity || 0).gt(
+    quote.input_amount_estimation
+  );
+
+  const bridgeFeeAmountInUsd =
+    scope === 'Bridge'
+      ? getCommonQuantity({
+          baseQuantity: quote.bridge_fee_amount,
+          chain: createChain(quote.input_chain),
+          asset: spendAsset,
+        }).times(spendAsset.price?.value || 0)
+      : null;
+
+  const priceImpact = calculatePriceImpact({
+    inputValue,
+    outputValue,
+    inputAsset: spendAsset,
+    outputAsset: receiveAsset,
+  });
+
+  const isHighPriceImpact = priceImpact && isHighValueLoss(priceImpact);
+  const outputAmountColor = isHighPriceImpact ? 'red' : 'grey';
+
+  return {
+    usd_amount_sent: toMaybeArr([usdAmountSend]),
+    usd_amount_received: toMaybeArr([usdAmountReceived]),
+    asset_amount_sent: toMaybeArr([quote.input_amount_estimation]),
+    asset_amount_received: toMaybeArr([quote.output_amount_estimation]),
+    asset_name_sent: toMaybeArr([spendAsset?.name]),
+    asset_name_received: toMaybeArr([receiveAsset?.name]),
+    asset_address_sent: toMaybeArr([spendAsset?.asset_code]),
+    asset_address_received: toMaybeArr([receiveAsset?.asset_code]),
+    gas: quote.transaction?.gas,
+    network_fee: null, // TODO
+    gas_price: null, // TODO
+    guaranteed_output_amount: quote.guaranteed_output_amount,
+    zerion_fee_percentage,
+    zerion_fee_usd_amount,
+    input_chain: quote.input_chain,
+    output_chain: quote.output_chain ?? quote.input_chain,
+    slippage: configuration.slippage,
+    contract_type: quote.contract_metadata?.name,
+    enough_balance,
+    enough_allowance: Boolean(quote.transaction),
+    warning_was_shown: isHighPriceImpact,
```

Comment:
Something is not right here
![Zerts Screenshot 2025-04-24 at 17 56 38@2x](https://github.com/user-attachments/assets/d8e12c83-2352-4bd5-9b5c-dd365d6eebd9)
But I can't understand what exactly

Reply:
should be fixed, just checked locally